src/profile-logic/import/chrome.js

```diff
@@ -678,10 +680,28 @@ async function processTracingEvents(
       }
 
       if (profileEvent.name === 'CpuProfile') {
-        profile.meta.profilingStartTime =
-          profileEvent.args.data.cpuProfile.startTime / 1000;
+        // Get profile start time from profile URL if available. Timestamp format is what
+        // is generated by the Chrome DevTools UI when saving profiles. Technically, that
+        // timestamp is of profile save instead of profile start, but one can generate a
+        // more accurate start timestamp through e.g. handling the Chrome DevTools
+        // Protocol `Profiler.consoleProfileStarted` event.
+        const timeStampRe =
+          /(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2}).*\.cpuprofile(?:$|\/|\?|#)/;
+        const match = timeStampRe.exec(profileUrl ?? '');
+        if (match) {
+          const dateTimeString = `${match[1]}-${match[2]}-${match[3]}T${match[4]}:${match[5]}:${match[6]}`;
+          const startTime = new Date(dateTimeString);
+          profile.meta.startTime = startTime.getTime();
+        } else {
+          profile.meta.startTime = 0;
+        }
+
+        // Assume profiling starts exactly on profile start time.
+        threadInfo.lastSeenTime = profile.meta.profilingStartTime = 0;
         profile.meta.profilingEndTime =
-          profileEvent.args.data.cpuProfile.endTime / 1000;
+          (profileEvent.args.data.cpuProfile.endTime -
+            profileEvent.args.data.cpuProfile.startTime) /
```

Comment:
it's not clear to me that the difference will be in milliseconds. It seems to me this may be tenths of milliseconds, but I'm not so sure.

Reply:
emm not sure whether I'm getting you here -- it's known that `cpuProfile.startTime` / `.endTime` is in microseconds, and one can convert both to milliseconds first before doing the difference (using `/ 1000`), but `a / 1000 - b / 1000` is mathematically equivalent to `(a - b) / 1000` via algebra, and the latter involves less operations.

If 1 `startTime` / `endTime` tick is not actually 1 microsecond though, I suppose there has to be a way to change this assumption.

Reply:
I would be amazed if there is a `startTime` / `endTime` that actually exceeds [`Number.MAX_SAFE_INTEGER`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) (that would probably be a profile that has a lifespan longer than the average human), but since we require only _millisecond_ precision, I suppose I should convert `startTime` / `endTime` to milliseconds first before doing the difference.

Reply:
0e626fdab5e7df7ca45e0fc98bc795d60499c4d3 converted `startTime` and `endTime` to milliseconds first but was reverted because I got a test `profilingEndTime` of 190.7039999961853 (due to floating point rounding error) instead of the correct 190.704.